src/components/menu/menu.component.js

```diff
@@ -1,132 +1,81 @@
-        };
-        filteredChildren.forEach(({ props }, i) => {
-          if (
-            props.children &&
-            getMenuText(props)
-              .toString()
-              .toLowerCase()
-              .startsWith(event.key.toLowerCase())
-          ) {
-            if (firstMatch === undefined) {
-              firstMatch = i;
-            }
-            if (i > actualFocusedItemIndex.current && nextMatch === undefined) {
-              nextMatch = i;
-            }
-          }
-        });
-
-        if (nextMatch !== undefined) {
-          setFocusToElement(undefined, nextMatch);
-        } else if (firstMatch !== undefined) {
-          setFocusToElement(undefined, firstMatch);
-        }
+      setFocusedItemIndex(newIndex);
+      if (submenuOpen) {
+        setOpenSubmenuIndex(newIndex);
       }
     },
-    [filteredChildren, setFocusToElement]
+    [children]
   );
 
-  let index = 0;
+  /* istanbul ignore next */
+  const onClickOutside = useCallback((event) => {
+    // Reset the state of the menu when clicking elsewhere
+    if (!Events.composedPath(event).includes(StyledMenuWrapper)) {
+      reset();
+      document.removeEventListener("click", onClickOutside);
+    }
+  }, []);
+
+  useEffect(() => {
+    document.addEventListener("click", onClickOutside);
 
-  return (
-    <StyledMenuWrapper data-component="menu" menuType={menuType}>
-      <StyledMenuItemsWrapper role="menubar">
-        {React.Children.map(children, (child) => {
-          const isFirstElement = index === 0;
-          const i = index;
+    return function cleanup() {
+      document.removeEventListener("click", onClickOutside);
+    };
+  });
 
-          if (child.type === MenuItem) {
-            index += 1;
-          }
+  return (
+    <StyledMenuWrapper
+      data-component="menu"
+      role="menubar"
+      menuType={menuType}
+      {...rest}
+    >
+      {React.Children.map(children, (child, index) => {
+        const isFirstElement = index === 0;
+        const isFocused = focusedItemIndex === index;
 
-          return (
-            <StyledMenuItem role="presentation" menuType={menuType}>
-              {React.cloneElement(child, {
-                menuType,
-                ...(child.type === MenuItem && {
-                  ref: menuItemsRefs.current[i],
-                  isFirstElement,
-                  menuItemIndex: i,
-                  isOpen: isOpenSubmenu === i,
-                  setIsOpenSubmenu,
-                  setFocusToElement,
-                  isOpenByArrowLeftOrRight,
-                  setOpenByArrowLeftOrRight,
-                  handleKeyDown: (ev) => handleKeyDown(ev, i),
-                }),
-              })}
-            </StyledMenuItem>
-          );
-        })}
-      </StyledMenuItemsWrapper>
+        return (
+          <MenuContext.Provider
+            value={{
+              menuType,
+              isFirstElement,
+              handleKeyDown: (ev, submenuOpen) =>
```

Comment:
Can this be simplified? Seems like we're passing it down, using `openSubmenu` but then passing it back as an argument to this function.
```suggestion
              handleKeyDown: ev =>
```

Reply:
I can't see a way - this is specifically for when you have a submenu item focus and you press the left or right key onto another submenu. Somehow one submenu needs to communicate to open the second submenu through Menu, and this was the cleanest way I could fine. 
Happy to change if there's a better way I can't see though!